realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -518,6 +560,82 @@ public void cancel() {
         }
     }
 
+    private class MakeOfferAsyncTask extends AsyncTask<String> {
```

Comment:
Its a custom async task implementing the logic for controlling listeners and what not. I'll rename to PermissionManagerAsynctask to make that more clear